src/dynd/types/struct_type.cpp

```diff
@@ -48,45 +50,17 @@ ndt::struct_type::~struct_type() {}
 intptr_t ndt::struct_type::get_field_index(const char *field_name_begin, const char *field_name_end) const
 {
   size_t size = field_name_end - field_name_begin;
-  if (size > 0) {
-    char firstchar = *field_name_begin;
-    intptr_t field_count = get_field_count();
-    const char *fn_ptr = m_field_names.cdata();
-    intptr_t fn_stride = reinterpret_cast<const fixed_dim_type_arrmeta *>(m_field_names.get()->metadata())->stride;
-    for (intptr_t i = 0; i != field_count; ++i, fn_ptr += fn_stride) {
-      const string *fn = reinterpret_cast<const string *>(fn_ptr);
-      const char *begin = fn->begin(), *end = fn->end();
-      if ((size_t)(end - begin) == size && *begin == firstchar) {
-        if (memcmp(fn->begin(), field_name_begin, size) == 0) {
-          return i;
-        }
-      }
+
+  for (intptr_t i = 0; i < m_field_count; ++i) {
+    const std::string &name = m_field_names[i];
+    if (name.compare(0, std::string::npos, field_name_begin, size) == 0) {
+      return i;
     }
   }
 
   return -1;
 }
 
```

Comment:
In the latest patch I just used ```std::string```.  It would be nice to have ```std::buffer``` for temporarily wrapping a ```const char *``` without making a copy, but it does not seem to exist.